### PR TITLE
Add api for GHES version

### LINKED ISSUE #753: Add ability to retrieve GHES version
I'm assuming there is a GHES endpoint we can call to retrieve this data. Need to implement this endpoint in GithubApi so we can use the version to decide whether to upload to cloud storage or not.

### PR DESCRIPTION
**Pull request by brianaj**
Fixes #753 

- [ ] Did you write/update appropriate tests
- [ ] Release notes updated (if appropriate)
- [ ] Appropriate logging output
- [ ] Issue linked
- [ ] Docs updated (or issue created)

### CODE REVIEW

File: src/OctoshiftCLI.Tests/GithubApiTests.cs [hunk 1]
```diff
@@ -2552,6 +2552,61 @@ public async Task UpdateSecretScanningAlert_Calls_The_Right_Endpoint_With_Payloa
...
+            var url = "https://api.github.com/meta";
+            var response = $@"
+            {{
+                ""verifiable_password_authentication"": true,
+                ""packages"": [
+                ],
+                ""dependabot"": [
+                ]
+            }}
+        ";
```
Comment (juanmrad) [inline src/OctoshiftCLI.Tests/GithubApiTests.cs:2568]:
```suggestion
            }}";
```

nit. same in line 2595 😄

File: src/OctoshiftCLI.Tests/GithubApiTests.cs [hunk 2]
```diff
@@ -2552,6 +2552,60 @@ public async Task UpdateSecretScanningAlert_Calls_The_Right_Endpoint_With_Payloa
...
+            var response = $@"
+            {{
+                ""verifiable_password_authentication"": true,
+                ""packages"": [
+                ],
+                ""dependabot"": [
+                ],
+                ""installed_version"": ""3.7.0""
+            }}
+        ";
```
Comment (juanmrad) [inline src/OctoshiftCLI.Tests/GithubApiTests.cs:2595]:
```suggestion
            }}";
```

this was missed 😅

Comment (brianaj) [reply src/OctoshiftCLI.Tests/GithubApiTests.cs:2595]: the `dotnet format` didn't do anything for me 😞

**Review by juanmrad — APPROVED** [review]

File: src/OctoshiftCLI.Tests/GithubApiTests.cs [hunk 3]
```diff
@@ -2552,6 +2552,59 @@ public async Task UpdateSecretScanningAlert_Calls_The_Right_Endpoint_With_Payloa
...
+            }}";
+
+            _githubClientMock
+                .Setup(m => m.GetAsync(url, null))
+                .ReturnsAsync(response);
+
+            // Act
+            var version = await _githubApi.GetEnterpriseServerVersion();
+
+
```
Comment (dylan-smith) [inline src/OctoshiftCLI.Tests/GithubApiTests.cs:2603]:
```suggestion
```

**Review by dylan-smith — APPROVED** [review]

**Pull request merged**